esp-hal/src/sdio.rs

```diff
@@ -0,0 +1,586 @@
+
+            w.sdio_slc0_rx_loop_test().clear_bit();
+            w.sdio_slc0_tx_loop_test().clear_bit()
+        });
+
+        slc.slcconf1().modify(|_, w| {
+            w.sdio_slc0_rx_stitch_en().clear_bit();
+            w.sdio_slc0_tx_stitch_en().clear_bit();
+
+            w.sdio_slc0_len_auto_clr().clear_bit()
+        });
+
+        slc.slc_rx_dscr_conf()
+            .modify(|_, w| w.sdio_slc0_token_no_replace().set_bit());
+
+        Ok(())
+    }
+
+    /// Sets the high-speed supported bit to be read by the host.
+    #[cfg(any(feature = "esp32", feature = "esp32c6"))]
+    fn pac_enable_hs(&mut self) -> Result<(), Error> {
+        self.hinf_block()
+            .cfg_data1()
+            .modify(|_, w| w.highspeed_enable().variant(self.config.hs()));
+
+        Ok(())
+    }
+
+    /// Sets the communication timing.
+    #[cfg(feature = "esp32")]
+    fn pac_set_timing(&mut self) -> Result<(), Error> {
+        let host = self.slchost_block();
+
+        match self.config.timing() {
+            Timing::PsendPsample => {
+                host.host_slchost_conf().modify(|_, w| unsafe {
+                    w.host_frc_sdio20().bits(0x1f);
+                    w.host_frc_sdio11().bits(0x0);
+                    w.host_frc_pos_samp().bits(0x1f);
+                    w.host_frc_neg_samp().bits(0x0)
+                });
+            }
+            Timing::PsendNsample => {
+                host.host_slchost_conf().modify(|_, w| unsafe {
+                    w.host_frc_sdio20().bits(0x1f);
+                    w.host_frc_sdio11().bits(0x0);
+                    w.host_frc_pos_samp().bits(0x0);
+                    w.host_frc_neg_samp().bits(0x1f)
+                });
+            }
+            Timing::NsendPsample => {
+                host.host_slchost_conf().modify(|_, w| unsafe {
+                    w.host_frc_sdio20().bits(0x0);
+                    w.host_frc_sdio11().bits(0x1f);
+                    w.host_frc_pos_samp().bits(0x1f);
+                    w.host_frc_neg_samp().bits(0x0)
+                });
+            }
+            Timing::NsendNsample => {
+                host.host_slchost_conf().modify(|_, w| unsafe {
+                    w.host_frc_sdio20().bits(0x0);
+                    w.host_frc_sdio11().bits(0x1f);
+                    w.host_frc_pos_samp().bits(0x0);
+                    w.host_frc_neg_samp().bits(0x1f)
+                });
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Sets the communication timing.
+    #[cfg(feature = "esp32c6")]
+    fn pac_set_timing(&mut self) -> Result<(), Error> {
+        let host = self.slchost_block();
+
+        match self.config.timing() {
+            Timing::PsendPsample => {
+                host.conf().modify(|_, w| unsafe {
+                    w.frc_sdio20().bits(0x1f);
+                    w.frc_sdio11().bits(0x0);
+                    w.frc_pos_samp().bits(0x1f);
+                    w.frc_neg_samp().bits(0x0)
+                });
```

Comment:
I think only this part needs to be chip specific, not the outer logic.

Reply:
Right, so instead create eight new functions guarded by conditional compilation? I think your suggestion to fix register naming in the PAC is probably the best route to take.

Reply:
What do you mean by 8? These 4 branches could be calls to the same function with 4 different sets of parameters (or a struct if you want them named), you need 2 variants that set the 4 register fields. That is one function per MCU, and 4 calls.

Reply:
> What do you mean by 8? These 4 branches could be calls to the same function with 4 different sets of parameters (or a struct if you want them named), you need 2 variants that set the 4 register fields. That is one function per MCU, and 4 calls.

Right, that's what the edited reply means. Just making two functions guarded by conditional compilation, and call the function four times. My initial response was mistaken.